### PR TITLE
add(kafka): enable mTLS for Kafka Connect and Debezium PostgreSQL

### PR DESCRIPTION
**Pull request by harshini-rangaswamy**
## Checklist

- [x] The first paragraph of the page is on one line.
- [x] The other lines have a line break at 90 characters.
- [x] I checked the output.
- [x] I applied the [style guide](../styleguide.md).
- [x] My links start with `/docs/`.

### CODE REVIEW

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:27]: Maybe we can also specify in case of cloud sql which are the steps to:
1. unlock the aiven ips (that are required to be whitelisted otherwise they get an error because the connection itsn't able to reach the db)
2. enable the 2 features I've mentioned initially in the presentation, the one for using the right level for the read only log (wal_level) and the other one for reading the log and being able to serialize/deserialize it from/into json. There are two different plugins I've mentioned only one in the slides if I'm not remembering badly they are `wal2json` and `pgoutput`

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:88]: oh the ordering seems wrong, from here on its always 1

**Review by eliax1996 — CHANGES_REQUESTED** [review]
Great, seems almost perfect :) let me merge the feature and check the set of commands before merging this one

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:88]: This is intentional. In our documentation setup, we maintain consistency by starting all numbered list items with 1 to facilitate easy insertion of steps. The numbering is adjusted during output generation.

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (MichaelTansiniAiven) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:27]: wal2json is about to be deprecated with the new DBZ 2.5 rollout so we should mention pgoutput and perhaps decoderbufs which are identified as the wal2json replacements in the docs

Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:27]: Add a subsection for CloudSQL and a note about wal2json deprecation.

**Review by eliax1996 — APPROVED** [review]
To me looks fine. Could we ask someone else to take a look and try to follow the instructions? IMO its important ask someone internal to attempt follow those and give a feedback after attempt using it

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (aiven-anton) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:127]:
```suggestion
   INTEGRATION_ENDPOINT_ID=$(
       avn service integration-endpoint-list --project $PROJECT \
           | grep external_postgresql \
           | awk '{print $1}'
   )
```

suggestion: Could be reformatted for readability.

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (aiven-anton) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:119]:
```suggestion
   avn service integration-endpoint-create \
       --project $PROJECT \
       -d external_postgresql \
       --user-config "$INTEGRATION_CONFIG"
```

suggestion: Formatting.

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md]: fixed

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md]: fixed

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (aiven-anton) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:123]:
```suggestion
	   --endpoint-name external_postgresql \
```

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (aiven-anton) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md]:
```suggestion
       --integration-type kafka_connect_postgresql \
```

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (aiven-anton) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md]:
```suggestion
	     --source-endpoint-id $INTEGRATION_ENDPOINT_ID \
```

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (aiven-anton) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md]:
```suggestion
       --dest-service <kafka_connect_name>
```

**Review by aiven-anton — COMMENTED** [review]
Added suggestions for using long-form argument names (for googlability and searchability).

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (MichaelTansiniAiven) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:39]: if these aren't already in place I assume

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (MichaelTansiniAiven) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:62]: use one of, not both

**Review by MichaelTansiniAiven — APPROVED** [review]
looks good to me but would check with @eliax1996 if hasn't been done already

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:39]: @eliax1996 Could you please confirm this? Based on our previous discussions and [this](https://github.com/aiven/aiven-docs/pull/38#discussion_r1482565924), added the section for CloudSQL.

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (ArthurFlag) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:84]: I'd argue we don't need to say the means to perform a task since we can now use tabs. 
Right now it's just cli, which doesn't bring SEO value here since it's the only choice, and when we get another possibility, it'll be in a tab, not in a new section.

```suggestion
## Configure the integration
```

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (ArthurFlag) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:194]: Shouldn't this be at the root of the section instead of being in the last step? Also wondering if `deploy the setup` (and how) shouldn't be a step itself? Your call, I don't how familiar users are with this.

**Review by ArthurFlag — CHANGES_REQUESTED** [review]

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:194]: Thank you, Arthur! I have replaced 'successfully deployed' with 'successfully configuring the setup' to clarify that our focus is on configuration, not deployment. I've included the steps in a dedicated verification section. Additionally, I have added a note about automatic topic creation in the 'create service' step, which is one of the first steps in the process.

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:84]: Sure. Updated the heading.

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:42]: yes @MichaelTansiniAiven its right, only if they aren't already in place

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:125]: `user` has been renamed to `username`

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:128]: we should add here also the `password`, the api has been changed and moved from the connector to here

Comment (eliax1996) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:128]: it should be another key value like: `"password": my_password`

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:168]: the comment should mention all the configurations provided by the external configuration, that could be: `"database.dbname","database.hostname","database.password","database.port","database.sslcert","database.sslkey","database.sslmode","database.sslrootcert","database.user"`

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:170]: this is provided by the `external_postgresql` integration

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:178]: maybe we can add also `"database.tcpKeepAlive": "true",`? So that by default the connection isn't closed if the connector isn't used for a while. Not sure if add it or not

Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:178]: those two transforms I think are optional, it depends on the usage of the user. Not sure if include it or just mention that could be used

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (eliax1996) [inline docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:169]: we should mention also `"topic.prefix": my_prefix`. And below we should say that the messages are published under a topic called: `"{prefix}.{database_name}.{replicated_table}"

**Review by eliax1996 — CHANGES_REQUESTED** [review]
Ok, last check done :) after that I think we are ready to merge!

**Review by ArthurFlag — APPROVED** [review]

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md:185]: Done.

Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md]: I have added them and mentioned that these are optional.

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md]: updated.

File: docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md
Comment (harshini-rangaswamy) [reply docs/products/kafka/kafka-connect/howto/kafka-connect-debezium-tls-pg.md]: Done

**Review by eliax1996 — APPROVED** [review]
Looks good :) proceeding with the merge

**Pull request merged**
